### PR TITLE
test(fix): add delay after cancel to prevent order error

### PR DESCRIPTION
**Pull request by RogerLamTd**

### CODE REVIEW

File: driver/chain_syncer/calldata/syncer_test.go [hunk 1]
```diff
@@ -79,6 +79,8 @@ func (s *CalldataSyncerTestSuite) TestCancelNewSyncer() {
 	defer s.cancel()
```
Comment (davidtaikocha) [inline driver/chain_syncer/calldata/syncer_test.go:79]: is this necessary? since i also see `cancel` in line 81

Comment (RogerLamTd) [reply driver/chain_syncer/calldata/syncer_test.go]: unnecessary, removed

File: driver/chain_syncer/calldata/syncer_test.go [hunk 2]
```diff
@@ -76,9 +76,10 @@ func (s *CalldataSyncerTestSuite) SetupTest() {
 	s.p = prop
 }
 func (s *CalldataSyncerTestSuite) TestCancelNewSyncer() {
-	defer s.cancel()
 	ctx, cancel := context.WithCancel(context.Background())
 	cancel()
+	// to ensure cancel is called
+	time.Sleep(1 * time.Second)
```
Comment (davidtaikocha) [inline driver/chain_syncer/calldata/syncer_test.go:82]: Can we use `<-ctx.Done()` here?

**Comment by davidtaikocha** [comment]
lets see if https://github.com/taikoxyz/taiko-client/pull/376 will fix this at first

**Pull request closed without merging**